docs/guide/developing/resources/create.md

```diff
@@ -0,0 +1,110 @@
+# Create
+
+## Description
+
+Device with collection resource allows dynamic creation of resources of selected type.
+
+## Guide
+
+To develop your own device you can examine the example in [cloud_server](https://github.com/iotivity/iotivity-lite/blob/master/apps/cloud_server.c). Lets examine the example to identify the necessary steps that allow a device to dynamically create resources.
+
+1. Create a collection resource
+```C
+  oc_resource_bind_resource_type(col, "oic.wk.col");
+```
+2. Determine which resource types can populate the collection
+```C
+  oc_collection_add_supported_rt(col, "oic.r.switch.binary");
```

Comment:
@jkralik, it seems only one of oc_collection_add_supported_rt or oc_collection_add_mandatory_rt is necessary. Not sure what is the difference according to docu:

```
oc_collection_add_supported_rt

Add a supported Resource Type to a collection

This will become the "rts" property of the collection. The "rts" property is
an array of Resource Types that are supported within an array of Links
exposed by the collection.
```

vs

```
oc_collection_add_mandatory_rt

Add a mandatory Resource Type to a collection

This will be come the "rts-m" property of the collection. The "rts-m"
property is an array of Resource Types that are mandatory to be exposed with
in an array of Links exposed by the collection.
```

At least one must be set, otherwise the create operation fails, but what's the point of it actually? In code it keeps supported and mandatory rts in a list and when a request comes and it doesn't have its rt property in one of these lists the operation fails.

Reply:
@Danielius1922  https://openconnectivity.org/specs/OCF_Core_Specification.pdf 7.8.2.8